### PR TITLE
DOC: improve SIMD features tables

### PR DESCRIPTION
**Pull request by seiko2plus**
**DOC: improve SIMD features tables**:
- improve the tables generator(style/simplify)
- show the differences between the compilers
- add an explanation about interrelated CPU features

### CODE REVIEW

File: doc/source/reference/simd/simd-optimizations-tables-diff.inc [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. table::
+    :align:left
+    =========== ======================================================================================================================
+    Name        Implies                                                                                                               
+    =========== ======================================================================================================================
+    ``FMA3``    ``SSE`` ``SSE2`` ``SSE3`` ``SSSE3`` ``SSE41`` ``POPCNT`` ``SSE42`` ``AVX`` ``F16C`` **``AVX2``**                      
+    ``AVX2``    ``SSE`` ``SSE2`` ``SSE3`` ``SSSE3`` ``SSE41`` ``POPCNT`` ``SSE42`` ``AVX`` ``F16C`` **``FMA3``**                      
+    ``AVX512F`` ``SSE`` ``SSE2`` ``SSE3`` ``SSSE3`` ``SSE41`` ``POPCNT`` ``SSE42`` ``AVX`` ``F16C`` ``FMA3`` ``AVX2`` **``AVX512CD``**
+    =========== ======================================================================================================================
```
Comment (mattip) [inline doc/source/reference/simd/simd-optimizations-tables-diff.inc:29]: Since the two intel compiler tables are the same, they can be combined.

Comment (seiko2plus) [reply doc/source/reference/simd/simd-optimizations-tables-diff.inc]: done

**Comment by mattip** [comment]
LGTM. [Here](https://14592-908607-gh.circle-artifacts.com/0/doc/build/html/reference/simd/simd-optimizations.html) is the new table

**Review by mattip — APPROVED** [review]

**Comment by seberg** [comment]
Looks fine to me as well, and CircleCI is happy. Thanks.

**Pull request merged**